components/clp-package-utils/clp_package_utils/scripts/native/del_archives.py

```diff
@@ -0,0 +1,147 @@
+import argparse
+import logging
+import pathlib
+import shutil
+import sys
+from contextlib import closing
+from typing import List
+
+from clp_py_utils.sql_adapter import SQL_Adapter
+
+from clp_package_utils.general import (
+    CLP_DEFAULT_CONFIG_FILE_RELATIVE_PATH,
+    get_clp_home,
+    load_config_file,
+)
+
+# Setup logging
+# Create logger
+logger = logging.getLogger(__file__)
+logger.setLevel(logging.INFO)
+# Setup console logging
+logging_console_handler = logging.StreamHandler()
+logging_formatter = logging.Formatter("%(asctime)s [%(levelname)s] [%(name)s] %(message)s")
+logging_console_handler.setFormatter(logging_formatter)
+logger.addHandler(logging_console_handler)
+
+
+def handle_file_deletion(
+    clp_home: pathlib.Path,
+    config_file_path: pathlib.Path,
+    default_config_file_path: pathlib.Path,
+    begin_ts: int,
+    end_ts: int,
+) -> int:
+    """
+    Deletes all archives with `begin_timestamp` and `end_timestamp` within the specified range from
+    the database, and removes any files associated with these archives.
+    :param clp_home:
+    :param config_file_path:
+    :param default_config_file_path:
+    :param begin_ts:
+    :param end_ts:
+    :return: 0 on success, -1 otherwise.
+    """
+
+    # Validate and load config file
+    try:
+        clp_config = load_config_file(config_file_path, default_config_file_path, clp_home)
+        clp_config.validate_logs_dir()
+    except:
+        logger.exception("Failed to load config.")
+        return -1
+
+    database_config = clp_config.database
+    archives_dir = clp_config.archive_output.directory
+    # Note, the error message doesn't output the value of archives_dir because it is a mounted
+    # path. It could be confusing for user because the path will not exist in their file system.
+    if not archives_dir.exists():
+        logger.error(f"archive directory doesn't exist. abort deletion")
+        return -1
+
+    archive_ids: List[str]
+    logger.info(f"Start deleting archives from database")
+    try:
+        sql_adapter = SQL_Adapter(database_config)
+        clp_db_connection_params = database_config.get_clp_connection_params_and_type(True)
+        table_prefix = clp_db_connection_params["table_prefix"]
+        with closing(sql_adapter.create_connection(True)) as db_conn, closing(
+            db_conn.cursor(dictionary=True)
+        ) as db_cursor:
+            db_cursor.execute(
+                f"""
+                DELETE FROM `{table_prefix}archives` WHERE
+                begin_timestamp >= {begin_ts} AND end_timestamp <= {end_ts}
+                RETURNING id
+                """
+            )
+            results = db_cursor.fetchall()
+
+            if 0 == len(results):
+                logger.warning("No archive falls into the specified time range, abort deletion")
```

Comment:
I also think this should be `info` since it's not really something that the user did wrong (there's no way they could really know what the time range of the archives are beforehand.